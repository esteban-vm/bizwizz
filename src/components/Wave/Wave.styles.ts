import type { BgColor } from '@/types'
import clsx from 'clsx'

const styles = (color: BgColor) => {
  return clsx('h-40 w-full', color === 'black' ? 'text-black' : color === 'white' ? 'text-white' : 'text-primary')
}

export default styles
