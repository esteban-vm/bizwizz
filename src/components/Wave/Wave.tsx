import type { FC, BgColor } from '@/types'
import clsx from 'clsx'

interface WaveProps {
  color: BgColor
}

const Wave: FC<WaveProps> = ({ color }) => {
  return (
    <svg
      className={clsx('wave', color === 'black' ? 'text-black' : color === 'white' ? 'text-white' : 'text-primary')}
      preserveAspectRatio='none'
      role='presentation'
      viewBox='0 0 500 150'
      aria-hidden
    >
      <path
        d='M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z'
        fill='currentColor'
        stroke='none'
      />
    </svg>
  )
}

export default Wave
