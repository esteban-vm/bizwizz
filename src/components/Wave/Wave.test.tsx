import { render, cleanup } from '@/tests'
import Wave from './Wave'

describe('Wave:', () => {
  afterEach(cleanup)

  it('should render correctly when is primary', () => {
    const { asFragment } = render(<Wave is='primary' />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly when is black', () => {
    const { asFragment } = render(<Wave is='black' />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly when is white', () => {
    const { asFragment } = render(<Wave is='white' />)
    expect(asFragment()).toMatchSnapshot()
  })
})
