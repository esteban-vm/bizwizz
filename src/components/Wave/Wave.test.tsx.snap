// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Wave: > should render correctly when is black 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="wave text-black"
    preserveAspectRatio="none"
    role="presentation"
    viewBox="0 0 500 150"
  >
    <path
      d="M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z"
      fill="currentColor"
      stroke="none"
    />
  </svg>
</DocumentFragment>
`;

exports[`Wave: > should render correctly when is primary 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="wave text-primary"
    preserveAspectRatio="none"
    role="presentation"
    viewBox="0 0 500 150"
  >
    <path
      d="M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z"
      fill="currentColor"
      stroke="none"
    />
  </svg>
</DocumentFragment>
`;

exports[`Wave: > should render correctly when is white 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="wave text-white"
    preserveAspectRatio="none"
    role="presentation"
    viewBox="0 0 500 150"
  >
    <path
      d="M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z"
      fill="currentColor"
      stroke="none"
    />
  </svg>
</DocumentFragment>
`;
